.github/workflows/ci-rerun.yml

```
@@ -0,0 +1,105 @@
+  workflow_dispatch:
+
+env:
+  GH_REPO: ${{ github.repository }}
+
+permissions:
+  contents: read
+  actions: write
+
+jobs:
+  scheduled-retry:
+    if: >
+      github.event_name == 'schedule' || github.event_name == 'workflow_dispatch'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Re-run CI workflow
+        run: |
+          PR_NUMBERS=$(gh pr list -l 'ci-retry' --json number --jq '.[].number')
+          for PR_NUMBER in $PR_NUMBERS; do
+            echo "Processing PR #$PR_NUMBER"
+            WORKFLOW_RUN_ID=$(gh pr view "$PR_NUMBER" --json headRefName --jq '.headRefName' | xargs -I {} gh run list --workflow=ci.yml --branch={} --json databaseId --jq '.[].databaseId')
+            if [ -n "$WORKFLOW_RUN_ID" ]; then
+              echo "Rerunning workflow run #$WORKFLOW_RUN_ID for PR #$PR_NUMBER"
+              gh run rerun "$WORKFLOW_RUN_ID"
+            else
+              echo "No runs found for PR #$PR_NUMBER"
+            fi
+          done
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  scheduled-retry-failed-jobs:
+    if: >
+      github.event_name == 'schedule' || github.event_name == 'workflow_dispatch'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Re-run CI workflow
+        run: |
+          PR_NUMBERS=$(gh pr list -l 'ci-retry-failed-jobs' --json number --jq '.[].number')
+          for PR_NUMBER in $PR_NUMBERS; do
+            echo "Processing PR #$PR_NUMBER"
+            WORKFLOW_RUN_ID=$(gh pr view "$PR_NUMBER" --json headRefName --jq '.headRefName' | xargs -I {} gh run list --workflow=ci.yml --branch={} --json databaseId --jq '.[].databaseId')
+            if [ -n "$WORKFLOW_RUN_ID" ]; then
+              echo "Rerunning workflow run #$WORKFLOW_RUN_ID for PR #$PR_NUMBER"
+              gh run rerun "$WORKFLOW_RUN_ID" --failed
+            else
+              echo "No runs found for PR #$PR_NUMBER"
+            fi
+          done
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  trigger-rerun:
+    if: >
+      (
+        github.event.label.name == 'ci-syntax-only' ||
+        github.event.label.name == 'ci-requeue' ||
+        github.event.label.name == 'ci-skip-install'
+      )
+    runs-on: ubuntu-latest
+    steps:
+      - name: Re-run CI workflow
+        run: |
+          PR_NUMBER=$(gh pr view "${{ github.event.pull_request.number }}" --json headRefName --jq '.headRefName')
+          WORKFLOW_RUN_ID=$(gh pr view "${{ github.event.pull_request.number }}" --json headRefName --jq '.headRefName' | xargs -I {} gh run list --workflow=ci.yml --branch={} --json databaseId,status --jq '.[].databaseId')
+          if [ -n "$WORKFLOW_RUN_ID" ]; then
+            echo "Rerunning workflow run #$WORKFLOW_RUN_ID for PR #$PR_NUMBER"
+            gh run rerun "$WORKFLOW_RUN_ID"
+          else
+            echo "No runs found for PR #$PR_NUMBER"
+          fi
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Let's avoid interpolating right into a shell script, see https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-an-intermediate-environment-variable.

```suggestion
          PR_NUMBER=$(gh pr view "$PULL_REQUEST_NUMBER" --json headRefName --jq '.headRefName')
          WORKFLOW_RUN_ID=$(gh pr view "$PULL_REQUEST_NUMBER" --json headRefName --jq '.headRefName' | xargs -I {} gh run list --workflow=ci.yml --branch={} --json databaseId,status --jq '.[].databaseId')
          if [ -n "$WORKFLOW_RUN_ID" ]; then
            echo "Rerunning workflow run #$WORKFLOW_RUN_ID for PR #$PR_NUMBER"
            gh run rerun "$WORKFLOW_RUN_ID"
          else
            echo "No runs found for PR #$PR_NUMBER"
          fi
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          PULL_REQUEST_NUMBER: ${{ github.event.pull_request.number }}
```
I also don't really follow what's going on with `PR_NUMBER` and `WORKFLOW_RUN_ID`. The content of `PR_NUMBER` won't be a pull request number, will it?